src/pykx/pandas_api/pandas_meta.py

```diff
@@ -311,3 +311,18 @@ def agg(self, func, axis=0, *args, **kwargs): # noqa: C901
             return data
         else:
             return (q('{(flip enlist[`function]!enlist x)!y}', keyname, data))
+
+    @api_return
+    def isna(self):
+        tab = self
+        return q.null(tab)
```

Comment:
What should we do here? (According to documentation: "Characters such as empty strings ... are not considered NA values (unless you set pandas.options.mode.use_inf_as_na = True).")
```python
>>> df
    Name  Age         City
0   John   25     New York
1  Alice   30  Los Angeles
2    Bob   22             
>>> df.isna()
    Name    Age   City
0  False  False  False
1  False  False  False
2  False  False  False
>>> kx.toq(df).isna()
pykx.Table(pykx.q('
Name Age City
-------------
0    0   0   
0    0   0   
0    0   1   
'))
```
Are there more cases like this? Perhaps the implementation is ok the way it is, but we should warn them about it or at least be prepared to discuss it.

Reply:
This is a conversation that at some point should come up and I am split between the two options: Should we adhere more to the pandas way of doing things or to the way we do things in q? I don't have a clear answer, although I think that since we are dealing with PyKX tables we should stick to q's philosophy.

However, it totally makes sense to keep the implementation as close to pandas as possible. The main argument would be that it would break compatibility with existing pandas code.

I have been looking into the pandas documentation and I have found that the pandas `Series` object (what I understand is the equivalent to `pykx.List`) also has these special cases on its `isna` method (see link below). Since the `pykx.q.null` method is already implemented then I consider that a decision has already been made: to keep the implementation close to q in terms of how null values are handled.

Of course I think this is still open to discussion. What do you think?

In any case, I think we should let the PyKX team know about this dilemma we are facing.

As for more cases like this, symbols seem to be affected as well on the q side. I would need to check if there are more affected cases.

https://pandas.pydata.org/docs/reference/api/pandas.Series.isna.html#pandas.Series.isna

Reply:
> This is a conversation that at some point should come up and I am split between the two options: Should we adhere more to the pandas way of doing things or to the way we do things in q?

Once we've gained more experience with the library, I'd say that we should go for the second option: the pandas API is more about syntax than it is about semantics. By embracing the q way of thinking, the resulting functions are way far simpler. So, I think that this family of functions are perfectly fine.